xwiki-platform-core/xwiki-platform-url/xwiki-platform-url-api/src/main/java/org/xwiki/url/URLSecurityManager.java

```diff
@@ -36,6 +36,15 @@
 @Role
 public interface URLSecurityManager
 {
+    /**
+     * Dedicated string used to escape {@code %} character.
+     * @see #parseToSafeURI(String)
+     * @since 15.1RC1
+     * @since 14.10.5
+     */
+    @Unstable
+    String PERCENT_ESCAPE = "__XWIKI_URL_SECURITY_PERCENT__";
```

Comment:
I'm not sure this should be part of the API as it's really an implementation detail that we need to use a placeholder because of the not-so-nice Java URI implementation.

Reply:
I agree that it's a technical detail, but making it an API is a good way to ensure we won't change it by accident and that it's publicly documented for any consumer of the API.

Reply:
> but making it an API is a good way to ensure we won't change it by accident

Changing it is not a problem as long as the tests are passing. I agree with @michitux that it's not a good idea to make this an API.

Reply:
> Changing it is not a problem as long as the tests are passing.

I don't agree, the purpose here to inform external consumer of the API about the token that should not be used in an URI: we don't have any test about that right now. As I mentioned in a comment, I'm hesitating to actually directly throw an error if such token is found. Do you think it's a better idea? If I use that protection, I could indeed avoid making that an API and enforce the behaviour with a test.

Reply:
> Do you think it's a better idea?

The best would be to support inputs with such values, but you decided it does not worth it because "the chance that this pattern is used somewhere are very low". Making this hack public make things worst because I'm sure one say we will want to improve this.

In the meantime, yes, producing an error for an input you don't support, which is the subject here, is always more clear.

Reply:
> In the meantime, yes, producing an error for an input you don't support, which is the subject here, is always more clear.

ok will do that